### PR TITLE
Refactor the builder 

### PR DESCRIPTION
**Pull request by ilija-lazoroski**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ where:
         -d/--dist-dir-path: The path where resulting Agent Plugin archive will be stored.
         Default: <current_working_directory>/dist
 
-        -s/--source-dir: The name of the source directory.
+        -s/--source-dir-name: The name of the source directory.
```
Comment (mssalvatore) [inline README.md:36]: We have `--dist-dir-path` and `--source-dir-name`. Do these take different kinds of input? If not, we should shorten them to `--dist-dir` and `--source-dir`.

File: tests/test_agent_plugin_build_options.py
Comment (mssalvatore) [inline tests/test_agent_plugin_build_options.py:1]: Add instructions on running the test suite to the README.md.

Comment (mssalvatore) [inline tests/test_agent_plugin_build_options.py:1]: I realize this is a draft PR at the moment. This is where we stand ATM: 
image

File: README.md (hunk 1 above)
Comment (ilija-lazoroski) [reply README.md:36]: Yep they do. `--dist-dir-path` gets path and `--source-dir-name` a string

File: tests/test_agent_plugin_build_options.py
Comment (ilija-lazoroski) [reply tests/test_agent_plugin_build_options.py:1]: Yep. Until now only `AgentPluginBuildOptions` has been tested so that kind of explains it.

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**